ui_framework/components/button/kui_button.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import classnames from 'classnames';
+
+import { KuiButtonLink } from './kui_button_link';
+import { KuiTooltip } from '../tooltip/kui_tooltip';
+
+export function KuiButton(props) {
+  if (props.href && props.onClick) {
+    throw new Error('Only href or onClick can be defined, not both');
+  }
+
+  if (props.hide) return null;
+
+  const classes = classnames('kuiButton', props.className);
+
+  if (props.href) {
+    return <KuiButtonLink {...props} />;
+  } else {
+    return <KuiTooltip text={ props.tooltip }>
+      <button
+        className={ classes }
+        aria-label={ props.tooltip }
+        onClick={ props.onClick }
+      >
+        { props.children }
+      </button>
+    </KuiTooltip>;
```

Comment:
What happens in the event that no tooltip is specified? Shouldn't we just return the element without wrapping it in `Tooltip`?

```javascript
export function KuiButton(props) {
  const classes = classnames('kuiButton', props.className);
  const button = (
    <button
      className={ classes }
      aria-label={ props.tooltip 
      onClick={ props.onClick }
    >
      { props.children }
    </button>
  );

  if (props.tooltip) {
    return <KuiTooltip text={ props.tooltip }>{button}</KuiTooltip>;
  } else {
    return button;
  }
}
```